src/pubkey.cpp

```diff
@@ -334,6 +336,38 @@ bool CPubKey::Derive(CPubKey& pubkeyChild, ChainCode &ccChild, unsigned int nChi
     return true;
 }
 
+std::optional<EllSqPubKey> CPubKey::EllSqEncode() const {
+    assert(secp256k1_context_verify && "secp256k1_context_verify must be initialized to use CPubKey.");
+
+    secp256k1_pubkey pubkey;
+    if (!secp256k1_ec_pubkey_parse(secp256k1_context_verify, &pubkey, vch, size())) {
+        return {};
+    }
+
+    uint8_t r32[32];
+    GetStrongRandBytes(r32, 32);
+
+    EllSqPubKey encoded_pubkey;
+    secp256k1_ellsq_encode(secp256k1_context_verify, encoded_pubkey.data(), r32, &pubkey);
+    return encoded_pubkey;
+}
+
+CPubKey::CPubKey(const EllSqPubKey& encoded_pubkey, bool compressed) {
```

Comment:
There is reason to support uncompressed public keys here I think.

Reply:
Done.